src/ui/pages/Backup/VerifyBackup.tsx

```diff
@@ -8,21 +8,45 @@ import ArrowLeftIcon from 'jsx:src/ui/assets/arrow-left.svg';
 import { zeroizeAfterSubmission } from 'src/ui/shared/zeroize-submission';
 import { UnstyledButton } from 'src/ui/ui-kit/UnstyledButton';
 import { Input } from 'src/ui/ui-kit/Input';
-import { clipboardWarning } from 'src/ui/pages/BackupWallet/clipboardWarning';
-import { useSizeStore } from '../useSizeStore';
-import { useMnemonicInput } from '../shared/useMnemonicInput';
-import * as helperStyles from '../shared/helperStyles.module.css';
-import { isSessionExpiredError } from '../shared/isSessionExpiredError';
-import { usePendingRecoveryPhrase } from '../shared/usePendingRecoveryPhrase';
+import { isSessionExpiredError } from 'src/ui/shared/isSessionExpiredError';
+import { useWindowSizeStore } from 'src/ui/shared/useWindowSizeStore';
+import { useMnemonicInput } from 'src/ui/shared/useMnemonicInput';
+import * as helperStyles from 'src/ui/features/onboarding/shared/helperStyles.module.css';
+import { useMutation } from '@tanstack/react-query';
+import { walletPort } from 'src/ui/shared/channels';
+import { invariant } from 'src/shared/invariant';
+import { useRecoveryPhrase } from './useRecoveryPhrase';
+import type { BackupContext } from './useBackupContext';
+import { useBackupContext } from './useBackupContext';
+import { clipboardWarning } from './clipboardWarning';
 
 const INPUT_NUMBER = 12;
 const ARRAY_OF_NUMBERS = Array.from({ length: INPUT_NUMBER }, (_, i) => i);
 
+async function verifyRecoveryPhrase(
+  context: BackupContext,
+  expected: string,
+  actual: string
+) {
+  if (context.appMode === 'onboarding') {
+    if (expected !== actual) {
+      throw new Error('Incorrect seed phrase');
+    }
+  } else {
+    const isCorrect = await walletPort.request('verifyRecoveryPhrase', {
```

Comment:
If I understand, we have expected value here in both contexts, so this helper can be simplified to the first condition

Reply:
Actually you're right, thanks! ✅